v2/pkg/protocols/http/request.go

```diff
@@ -391,9 +391,48 @@ func (request *Request) executeRequest(reqURL string, generatedRequest *generate
 			}
 		}
 		if resp == nil {
+			switch request.Signature.Value {
```

Comment:
we can move this code to a separate function

Reply:
done